ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -168,6 +169,7 @@ private PeerDiscoveryController(
     this.outboundMessageHandler = outboundMessageHandler;
     this.peerBondedObservers = peerBondedObservers;
     this.discoveryProtocolLogger = new DiscoveryProtocolLogger(metricsSystem);
+    this.bondingPeers = new HashMap<>();
```

Comment:
I created the new collection so that the BONDING nodes did not interfere with the list of peers, specifically in regards to when the bucket is full and things need to be removed.